olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.h

```diff
@@ -60,6 +61,12 @@ class PartitionsRepository {
   static model::Partition PartitionFromSubQuad(
       const model::SubQuad& sub_quad, const std::string& partition);
 
+  static PartitionsResponse GetAggregatedTile(
+      const client::HRN& catalog, const std::string& layer,
+      client::CancellationContext cancellation_context,
+      const TileRequest& request, boost::optional<int64_t> version,
+      const client::OlpClientSettings& settings, bool aggregated);
```

Comment:
I don't think that the GetAggregatedTile should give you back something else than an aggregated tile. Having an additional parameter `bool aggregated` doesn't make much sense.
A more generic solution you found, almost, with the FindPartition() which can be used by both GetAggregatedTile() and GetPartitionForVersionedTile() or others who use or should use the QuadTreeIndex.
What is missing is a common solution for finding the correct quadtree index in the cache, which currently is in GetAggregatedTile() only. But let's postpone that to another time.